packages/docs/docs/guides/3-Portal-API/2-payment-status.md

```diff
@@ -3,52 +3,59 @@ title: Payment status with the Portal API
 sidebar_label: Payment status
 keywords:
 description: Learn how to integrate Request network and its features.
-
 ---
 
-After creation, each request has a payment status, you can view the metadata of a request which contains the payment status via:
+After creation, each request has a payment status, you can view the data of a request which contains the payment status via:
 
 ```jsx
-const requestData = request.getData();
+const result = await axios.get(`https://api.request.network/requests/${requestId}`);
+const request = result.data.request;
 ```
 
-You will receive back an object that looks like this: 
+You will receive back an object that looks like this:
+
 ```jsx
 /*
 { 
   balance,
   state,
   expectedAmount,
+  ...
 }
 */
 ```
 
-To get the payment status of a Request you can use the requestData object to check if the balance is greater than or equal to the expectedAmount. 
+To get the payment status of a Request you can use the requestData object to check if the balance is greater than or equal to the expectedAmount.
 
 If the balance >= expectedAmount - this means the request is paid.
 If the balance > 0 but < expectedAmount - this means the request is partially paid.
 If the balance == 0 - this means the request is unpaid.
 
-You can use the following snippet to see if the request has been paid. 
+You can use the following snippet to see if the request has been paid.
 
 ```jsx
 // Import Big Number package
-const BN = require('bn.js')
-
+const BN = require('bn.js');
 (async () => {
   // Check the balance of the request
   const requestData = request.getData();
-  const balance = requestData.balance;
-  console.log(`Balance of the request in ETH: ${balance}`);
-  
+  const balanceObject = requestData.balance;
+
+  if (balanceObject?.error) {
+    console.error(balanceObject.error.message);
+    return;
```

Comment:
2 comments here:  we should handle case when balanceObject is not set insteady of using `!`, and I'm not sure there's a `message` field on `balanceObject.error`... isn't it plain text?

```suggestion
  if (!balanceObject){
    console.error("balance not set");
    return;
  }
  if(balanceObject.error) {
    console.error(balanceObject.error);
    return;
```

Reply:
```
/** Interface for error encounter when getting the balance */
export interface IBalanceError {
  message: string;
  code: BALANCE_ERROR_CODE;
}
```

Reply:
ok ! my bad